p2p/exchange.go

```diff
@@ -157,8 +167,15 @@ func (ex *Exchange[H]) Head(ctx context.Context, opts ...header.HeadOption[H]) (
 	)
 	for _, from := range peers {
 		go func(from peer.ID) {
+			_, newSpan := tracerClient.Start(
+				ctx, "requesting peer",
+				trace.WithAttributes(attribute.String("peerID", from.String())),
+			)
+			defer newSpan.End()
+			
 			headers, err := ex.request(reqCtx, from, headerReq)
```

Comment:
So to chain spans we would pass the ctx from Start to request. 
But as long as we don't have spans in request, this is not necessary